pkg/server/commands/listusers/list_users_rpc.go

```diff
@@ -415,24 +431,30 @@ LoopOnIterator:
 			rewrittenReq := req.clone()
 			rewrittenReq.Object = &openfgav1.Object{Type: userObjectType, Id: userObjectID}
 			rewrittenReq.Relation = userRelation
-			return l.expand(ctx, rewrittenReq, foundUsersChan)
+			resp := l.expand(ctx, rewrittenReq, foundUsersChan)
+			if resp.hasCycle {
+				hasCycle = true
+			}
```

Comment:
nit: can simplify `hasCycle = hasCycle || resp.hasCycle`

Reply:
Interesting enough, that causes a data race.

Reply:
Oh! You have the data race even without my suggestion. `var hasCycle bool` should be an `atomic.Bool`